pkg/certificate/providers/vault/certificate_manager_test.go

```diff
@@ -114,99 +48,115 @@ var _ = Describe("Test client helpers", func() {
 			Expect(actual).To(Equal(expected))
 		})
 	})
+})
+
+func Test_New(t *testing.T) {
+	core, token, addr := mockVault(t)
+	defer core.Shutdown()
 
-	Context("Test Hashi Vault functions", func() {
-		cm := CertManager{
-			ca: rootCert,
-		}
-		cm.cache.Store(expiredCertCN, expiredCert)
-		cm.cache.Store(validCertCN, validCert)
-
-		getCachedCertificateCNs := func() []certificate.CommonName {
-			var commonNames []certificate.CommonName
-			cm.cache.Range(func(cnInterface interface{}, certInterface interface{}) bool {
-				cert := certInterface.(*certificate.Certificate)
-				commonNames = append(commonNames, cert.GetCommonName())
-				return true // continue the iteration
-			})
-			return commonNames
-		}
-
-		It("gets issuing CA public part", func() {
-			certSerialNum := certificate.SerialNumber(uuid.New().String())
-			expectedNumberOfCertsInCache := 2
-			Expect(len(getCachedCertificateCNs())).To(Equal(expectedNumberOfCertsInCache))
-			Expect(getCachedCertificateCNs()).To(ContainElement(certificate.CommonName("this.has.expired")))
-			Expect(getCachedCertificateCNs()).To(ContainElement(certificate.CommonName("valid.certificate")))
-			certBytes := uuid.New().String()
-			issue := func(certificate.CommonName, time.Duration) (*certificate.Certificate, error) {
-				cert := &certificate.Certificate{
-					IssuingCA:    pem.RootCertificate(certBytes),
-					SerialNumber: certSerialNum,
+	testCases := []struct {
+		description string
+		wantErr     bool
+	}{
+		{description: "no err", wantErr: false},
+		{description: "all nil", wantErr: true},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.description, func(t *testing.T) {
+			if tc.wantErr {
```

Comment:
doesn't look like. your testing the success case, so table driven tests aren't needed here.

Typically in a TDT, you would add the inputs to the test case struct.

Reply:
`doesn't look like. your testing the success case` - this was an accident 

Yeah,  I was flip flopping around on how to test this in a normal way. I could just add input and actually test what happens if we send in stuff, which _could_ be more useful than "nil" and "not nil". If you don't think so though I can take out the tables.